### DISCUSSION TITLE
Any suggestions for poor data of Madagascar and Sierra Leone?

### DISCUSSION THREAD

**Discussion by Haiping007** [pypsa-meets-earth/pypsa-earth#922] [Q&A]
Hi, 
Besides Kenya, we also try to run Pypsa-earth for Madagascar and Sierra Leone.
But we found the data of generators, lines and substations (pypsa-earth\resources\osm\raw) are poor, compared with the grid map in some local reports. 

For example, we plot the grid map of osm raw data for Madagascar from pypsa-earth:
![border_osm_region](https://github.com/pypsa-meets-earth/pypsa-earth/assets/146131420/528bb79a-44bf-401a-91d4-3a0296afa5c5)

And we found there are more existing transmission lines in the LEAD project report of Madagascar:
![Least-Cost Electricity Access Development Project (LEAD) in Madagascar](https://github.com/pypsa-meets-earth/pypsa-earth/assets/146131420/d3e0a2de-d1e4-4ed1-86e1-c69edc21f862)

Besides, we found the future energy demand of MG and SL (pypsa-earth\data\ssp2-2.6) are 0 in 2030 and 2050.  

We are planning to do power system modelling optimization for Madagascar and Sierra Leone in the year of 2030 and 2050. We have contacted the locals and tried to collect the detailed data, but it seems difficult. 

Do you have any suggestions on the solutions for the poor data?
Thanks!

**Comment by davide-f** [comment]
Hello!
Thanks for reporting!
Indeed there are few countries with empty data or limited information.
That is mostly due to the limited available data for the country and the fast pace at which developments are made that are not reflected into the databases.

For such cases, in the following, you can find some recommendations:
- regarding lines, the best is to contribute back to open-street-map the data that you found. That can help the whole global community even beyond the pypsa-earth one
- for the time-being and fast action, you could stream custom data to populate the missing information by OSM and alike. The recent PR by @carlosfv92 has enabled that (https://github.com/pypsa-meets-earth/pypsa-earth/pull/842). In the config file, under clean_osm_options, you can see the custom options that provide additional input files for the above. You just need to create additional files with the same format as the raw osm files available in resources/osm/raw
- For investigating microgrids cases, probably, you may be interested in the pypsa-distribution package that we are pushing. If interested every 2 weeks, there is the pypsa-distribution meeting at 5PM CEST on Friday. The next one is this friday (17th)

Apologies for the delay, but I was out of office for a while for working reasons, with limited connection

**Reply by Haiping007** [reply]
Hi, thank you for your reply! @davide-f 
I will take a look and try to figure out how to conduct your recommendations.
Best, 
Haiping

**Reply by carlosfv92** [reply]
Hey @Haiping007! The pull request Davide has linked has a bunch of information about how to include custom information if OSM data is not enough, so if you need any help with the configuration or how it works, just tag me in the conversation. 
However! I would say that the first thing is to find/get the information you want to include as this is the data to be provided to the model.

**Reply by Haiping007** [reply]
Hi! @carlosfv92 Thank you! Yes, I am still collecting data from Madagascar. It seems a time-consuming process and not easy......I will let you know once I get them.
Best!
